File: frontend/src/models/car/CarDetailsDto.ts
import {UserOwnerOverviewDto} from "../user/UserOwnerOverviewDto.ts";

export type CarDetailsDto = {
    id: string,
    category: string,
    brand: string,
    model: string,
    year: string,
    mileage: string,
    transmission: string,
    power: string,
    fuelType: string,
    price: string,
    imageUrl: string,
    description: string,
    fuelConsumption: string,
    numberOfDoors: string,
    numberOfSeats: string,
    airBags: string,
    numberOfPreviousOfOwners: string,
    color: string,
    owner: UserOwnerOverviewDto
}

File: backend/src/main/java/org/example/backend/model/CarDetailsDto.java
package org.example.backend.model;

public record CarDetailsDto(
        String id,
        String category,
        String brand,
        String model,
        String year,
        String mileage,
        String transmission,
        String power,
        String fuelType,
        String price,
        String imageUrl,
        String description,
        String fuelConsumption,
        String numberOfDoors,
        String numberOfSeats,
        String airBags,
        String numberOfPreviousOfOwners,
        String color,
        UserOwnerOverviewDto owner
) {
}